src/test/java/org/opentripplanner/routing/street/CarRoutingTest.java

```diff
@@ -0,0 +1,73 @@
+package org.opentripplanner.routing.street;
+
+import static org.opentripplanner.PolylineAssert.assertThatPolylinesAreEqual;
+
+import java.time.Instant;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.ConstantsForTests;
+import org.opentripplanner.model.GenericLocation;
+import org.opentripplanner.routing.algorithm.mapping.GraphPathToItineraryMapper;
+import org.opentripplanner.routing.api.request.RoutingRequest;
+import org.opentripplanner.routing.core.TraverseMode;
+import org.opentripplanner.routing.core.TraverseModeSet;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.impl.GraphPathFinder;
+import org.opentripplanner.standalone.config.RouterConfig;
+import org.opentripplanner.standalone.server.Router;
+
+public class CarRoutingTest {
+
+    static long dateTime = Instant.now().toEpochMilli();
+
+    /**
+     * The OTP algorithm tries hard to never visit the same node twice. This is generally a good
+     * idea because it avoids useless loops in the traversal leading to way faster processing time.
+     * <p>
+     * However there is are certain rare pathological cases where through a series of turn
+     * restrictions and roadworks you absolutely must visit a vertex twice if you want to produce a
+     * result. One example would be a route like this: https://tinyurl.com/ycqux93g (Note: At the
+     * time of writing this Hindenburgstr. (https://www.openstreetmap.org/way/415545869) is closed
+     * due to roadworks.)
+     * <p>
+     * This test checks that such a loop is possible.
+     * <p>
+     * More information: https://github.com/opentripplanner/OpenTripPlanner/issues/3393
+     */
+    @Test
+    @DisplayName("car routes can contain loops (traversing the same edge twice)")
+    public void shouldAllowLoopCausedByTurnRestrictions() {
+        var hindenburgStrUnderConstruction = ConstantsForTests.buildOsmGraph(
+                ConstantsForTests.HERRENBERG_HINDENBURG_STR_UNDER_CONSTRUCTION_OSM);
```

Comment:
Egyptian: (), break end down.

Reply:
I used the Intellij code style supplied by you to automatically format this file. Is this not the result you were expecting?